src/screens/Profile/Profile.js

```diff
@@ -805,6 +829,38 @@ class Profile extends React.Component<Props, State> {
             >
               <SystemInfoModal headerOnClose={() => this.setState({ showSystemInfoModal: false })} />
             </SlideModal>
+
+            <SlideModal
+              isVisible={showJoinBetaModal}
+              fullScreen
+              showHeader
+              backgroundColor={baseColors.snowWhite}
+              avoidKeyboard
+              title="join beta"
+              onModalHide={() => this.setState({ showJoinBetaModal: false })}
+            >
+              <StyledWrapper regularPadding flex={1}>
+                <Description>
+                  {
+                    'By joining the beta program, you will be added to our Firebase Analytics data collection.' +
+                    'Through this, Pillar will collect your username in order to enable beta features and monitor ' +
+                    'your wallet experience for any bugs and/or crashes while testing the new functionality. ' +
+                    'You can opt out of the beta program and Firebase Analytics collection at any time ' +
+                    'via the "System" under Settings.'
+                  }
+                </Description>
+                <Button
+                  title="Join Beta Testing"
+                  onPress={() => {
+                    setUserJoinedBeta(true);
+                    this.setState({ showJoinBetaModal: false });
```

Comment:
I'd move this to a single function call.

Reply:
Had to refactor this a bit as success toast was appearing in modal which closed right after. Please take a look.